e2e/cypress/integration/enterprise/system_console/openid/openid_spec.js

```diff
@@ -0,0 +1,235 @@
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Group: @enterprise @system_console
+
+import * as TIMEOUTS from '../../../../fixtures/timeouts';
+import {hexToRgbArray, rgbArrayToString} from '../../../../utils';
+
+const FAKE_SETTING = '********************************';
+const SERVICE_PROIVDER_LABEL = 'Select service provider:';
+const DISCOVERY_ENDPOINT_LABEL = 'Discovery Endpoint:';
+const CLIENT_ID_LABEL = 'Client ID:';
+const CLIENT_SECRET_LABEL = 'Client Secret:';
+const OPENID_LINK_NAME = 'OpenID Connect';
+const SAVE_BUTTON_NAME = 'Save';
+const CONVERT_NAME = 'Convert to OpenID Connect';
+const OAUTH2_NAME = 'OAuth 2.0deprecated';
+
+// # Goes to the System Scheme page as System Admin
+const goToAdminConsole = () => {
+    cy.apiAdminLogin();
+    cy.visit('/admin_console');
+};
+
+const verifyOAuthLogin = (id, text, color, href) => {
+    cy.apiLogout();
+
+    cy.findByRole('link', {name: `${text}`}).then((btn) => {
+        expect(btn[0].href).equal(href);
+        if (color) {
+            const rbgArr = hexToRgbArray(color);
+            expect(btn[0].style.backgroundColor).equal(rgbArrayToString(rbgArr));
+        }
+    });
+};
+
+describe('MM-27688 - System console-OpenId Connect', () => {
+    before(() => {
+        // * Check if server has license
+        cy.apiRequireLicense();
+    });
+    it('MM-27688 - Set to Generic OpenId', () => {
+        // # Go to admin console and set permissions as listed in the test
+        goToAdminConsole();
+
+        cy.findByRole('link', {name: OPENID_LINK_NAME}).click();
+
+        // # Click the OpenId header dropdown
+        cy.wait(TIMEOUTS.FIVE_SEC);
+
+        cy.findByLabelText(SERVICE_PROIVDER_LABEL).select('openid').wait(TIMEOUTS.ONE_SEC);
+
+        cy.findByLabelText('Button Name:').clear().type('TestButtonTest');
+
+        cy.get('#OpenIdSettings\\.ButtonColor-inputColorValue').clear().type('#c02222');
+
+        cy.findByLabelText(DISCOVERY_ENDPOINT_LABEL).clear().type('http://test.com/.well-known/openid-configuration');
+        cy.findByLabelText(CLIENT_ID_LABEL).clear().type('OpenIdId');
+        cy.findByLabelText(CLIENT_SECRET_LABEL).clear().type('OpenIdSecret');
+
+        cy.findByRole('button', {name: SAVE_BUTTON_NAME}).click().wait(TIMEOUTS.ONE_SEC);
+
+        // * Get config from API
+        cy.apiGetConfig().then(({config}) => {
+            expect(config.OpenIdSettings.Secret).to.equal(FAKE_SETTING);
+            expect(config.OpenIdSettings.Id).to.equal('OpenIdId');
+            expect(config.OpenIdSettings.DiscoveryEndpoint).to.equal('http://test.com/.well-known/openid-configuration');
+        });
+
+        verifyOAuthLogin('OpenIdButton', 'TestButtonTest', '#c02222', 'http://localhost:8065/oauth/openid/login?extra=expired');
```

Comment:
Please replace `http://localhost:8065` with `Cypress.config('baseUrl')` so we can run this against remote test server. Same with other applicable instance.